src/zenml/integrations/s3/flavors/s3_artifact_store_flavor.py

```diff
@@ -106,6 +113,40 @@ def _validate_client_kwargs(
             )
         return value
 
+    @property
+    def bucket(self) -> str:
+        """The bucket name of the artifact store.
+
+        Returns:
+            The bucket name of the artifact store.
+        """
+        if self._bucket is None:
+            self._bucket = os.path.normpath(self.path).split(os.sep)[1]
+        return self._bucket
+
+    @property
+    def is_versioned(self) -> bool:
```

Comment:
I think this would be better on the artifact store object itself, not the config: Usually our convention is that the config objects work without importing the client libraries of the integration, so other developers might think that they can use this property but it will crash if the integration is not installed.

Reply:
Moved it and also moved it to the init method, so the user will see the warning message early and not somewhere in the middle of pipeline logs.